notebooks/page view functional network/page_view_approach_economic_recovery.ipynb

```diff
@@ -0,0 +1,1508 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Exploring the economic recovery whole user journey subgraph"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This approach does not rely on the existing knowledge graph. It uses a functional graph based on page hit session data, to find a list of pages related to the economic recovery whole user journey (WUJ).\n",
+    "\n",
+    "ASSUMPTIONS: \n",
+    "- A set of pre-defined pages have been removed from the economic recovery WUJ subgraph\n",
+    "- Any pages with less than 10 session hits are removed from the economic recovery WUJ subgraph\n",
+    "- Any pages with a shortest page path length equal to or greater than 3 from both `seed0` pages are removed from the economic recovery WUJ subgraph\n",
+    "- Any pages where accumulated edge weight is equal to or lower than 20 are removed from the economic recovery WUJ subgraph\n",
+    "- Any browse pages not related to the economic recovery WUJ are manually removed from the economic recovery WUJ subgraph\n",
+    "\n",
+    "OUTPUT: \n",
+    "- A csv containing a list of pages related to the economic recovery WUJ. Sorted in ascending order in relation to a composite metric made up of: shortest page path from both `seed0` pages, and degree centrality.\n",
+    "  - Additional columns to filter on are included: `document type`, `document supertype`, `number of sessions that visit this page`, `number of sessions where this page is an entrance hit`, `number of sessions where this page is an exit hit`, `distance from /browse/working/finding-job`, `distance from /topic/further-education-skills/apprenticeships`, `the number of pages a user moves to/from between this page and another page in the list`\n",
+    "\n",
+    "REQUIREMENTS: \n",
+    "- Run `step_one_identify_seed_pages.ipynb` to define `seed0` and `seed1` pages\n",
+    "- Run `step_two_extract_page_hits.sql` to extract page hits for sessions that visit at least one `seed0` or `seed1` page\n",
+    "- Run `step_three_extract_nodes_and_edges.sql` to extract nodes and edges \n",
+    "- Run `step_four_create_networkx_graph.ipynb` to create NetworkX graph of the economic recovery whole user journey"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Import statements  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from neo4j import GraphDatabase\n",
+    "import networkx as nx\n",
+    "import os\n",
+    "import pandas as pd\n",
+    "import matplotlib.pyplot as plt\n",
+    "from tqdm.notebook import tqdm\n",
+    "import operator\n",
+    "from operator import itemgetter\n",
```

Comment:
You import operator and itemgetter from operator, but later on in your code, you write operator.itemgetter, when you could have just written itemgetter without the operator part before it. See a comment later on for more detail. Besides this, you don't use operator for anything else, so you should just be able to delete 'import operator' and keep from operator import itemgetter'

Reply:
Thanks for spotting this Jake! Resolved in commit 287f84a